File: utilities/elixir/parse.ex
# Parsing utilities for Elixir solutions

defmodule Parse do
  # Parse integers from a line of text
  def parse_ints(line) do
    line
    |> String.split(~r/-?\d+/, include_captures: true)
    |> Enum.filter(&Regex.match?(~r/^-?\d+$/, &1))
    |> Enum.map(&String.to_integer/1)
  end
end

File: utilities/rust/parse.rs
/// Parsing utilities for Rust solutions.

/// Parse integers from a line of text.
/// Splits the line by whitespace and extracts all integers.
pub fn parse_ints(line: &str) -> Vec<i64> {
    line.split_whitespace()
        .filter_map(|s| s.parse().ok())
        .collect()
}
